src/range.rs

```diff
@@ -182,6 +182,69 @@ pub fn take_while1<I, F>(f: F) -> TakeWhile1<I, F>
     TakeWhile1(f, PhantomData)
 }
 
+pub struct TakeUntil<I>(I::Range) where I: RangeStream;
+impl<I> Parser for TakeUntil<I>
+    where I: RangeStream,
+          I::Range: PartialEq + ::primitives::Range
+{
+    type Input = I;
+    type Output = I::Range;
+
+    #[inline]
+    fn parse_lazy(&mut self, mut input: Self::Input) -> ConsumedResult<Self::Output, Self::Input> {
+        use ::primitives::Range;
+
+        let len = self.0.len();
+        let mut to_consume = 0;
+        let mut look_ahead_input = input.clone();
+    
+        loop {
+            match look_ahead_input.clone().uncons_range(len) {
+                Ok(xs) => {
+                    if xs == self.0 {
+                        break;
+                    } else {
+                        to_consume += 1;
+                        if look_ahead_input.uncons().is_err() {
+                            unreachable!();
+                        }
+                    }
+                },
+                Err(_) => return EmptyErr(ParseError::empty(look_ahead_input.position())),
```

Comment:
Isn't the point of `parse_lazy` to add empty errors which can be filled in later? Or am I misunderstanding?